src/cli.rs

```diff
@@ -999,18 +1012,11 @@ async fn repl<'a, L: 'static + Ledger, C: CLI<'a, Ledger = L>>(
                     .replace('/', "_")
                     .replace('\\', "_")
             ));
-            (dir, None)
-        }
-        None => {
-            let tmp_dir = TempDir::new("keystore").context(IoSnafu)?;
-            (PathBuf::from(tmp_dir.path()), Some(tmp_dir))
+            dir
         }
+        Some(storage) => return Ok((keystore_for_test::<L, C>(storage, args).await?, reader, io)),
```

Comment:
In the old version, the only difference in this case (when `use_tmp_storage`) is that we initialize the wallet with a fresh, temporary path, rather than a possibly existing path. In the new version, `--use-tmp-storage` causes us to use an entirely different loader, and one only meant for test code, at that. Why?

Reply:
I removed all this test code from the cli.  It wasn't actually needed.  Instead I just have the test use the same loader (`Loader`) as the cli normally would. 

I think when I made this change originally I assumed we were only using the temp storage in tests but that was a mistake.